bloom-housing/bloom
# refactor: flatten translations strings, separate partners strings

# Pull Request Template

## Issue Overview

This PR addresses #2514


Remaining work: jurisdiction uptake, removing all plaintext strings from partners

## Description

* Flattened all of our translations files from their object format for usability - added a script to do this so we can use it in jurisdictions
* Separated out strings that are only used in partners to the partners english strings override file (did this manually 😭)
* Removed unused keys that do not have translations, kept in unused keys that do have translations
* Removed duplicate translations where possible (some are duplicated across preferences, where we are dynamically getting the key based on the preference prefix, so removing these duplicates would be a bigger change) - added a script to find duplicate values
* Updated the missing translations script to account for the new flattened format (here are the [missing public strings](https://docs.google.com/spreadsheets/d/1B063YeMOt2vakl1uZxsLvbAI8EwfbdV8B_z_uNKtiJw/edit#gid=0) off of this PR)
* Removed a few plaintext strings from partners, but my recommendation is to do this as we go from now on as we see them
* Added a script to check if there are keys in foreign translation files that do not exist in the english file

Uptaking this in jurisdictions might be a little tricky since there are some jurisdictionally specific strings that may have been added to the general file. My recommendation would be to run the flatten script over there, pull out the same strings to the partners file based on our core partners overrides, and check for differences to see if there are any strings added to the english file. Could be a little tedious, but we shouldn't just keep the same core files without checking.

## How Can This Be Tested/Reviewed?

I think this might be pretty challenging to review. Since pulling out the partners strings required lots of manual checking, it's possible I moved a string into the partners overrides file that is still used in public. I did run through all the strings twice, but that seems challenging to test. You could review the flatten script and ensure it's pulling all the strings through.

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
